Allow negative logic for the `is` operator.

galegosimpatico opened the pull request:
Where `variable is type` construct is allowed, allow the `variable is not type` construct too.

Please handle this with the required care and calm... but I think it should be mostly alright.

Safety: **`frogatto` not checked**, `citadel` booting, `anura` tests passing.

Cheers and Regards,

<img src="http://linuxgamingnews.org/wp-content/uploads/2010/08/airplane_boss.png" alt="frogatto plane" width="480"/>

galegosimpatico committed:
Allow negative logic for the `is` operator.

Where the `variable is type` construct is allowed, allow the `variable
is not type` construct too.

davewx7 commented on src/formula.cpp:
```diff
@@ -4773,6 +4779,11 @@ static std::string debugSubexpressionTypes(ConstFormulaPtr & fml)
 			int consume_backwards = 0;
 			std::string op_name(op->begin,op->end);
 
+			if (op_name == "is" && op > i1 && op + 1 > i1 &&
```
You don't need to check op + 1 > i1 since op > i1 already guarantees this, however you *should* check that op + 1 < i2 before using op + 1 to make sure we don't run off the end of the buffer.

davewx7 reviewed (changes requested):
Looks great other than my one comment.

galegosimpatico committed:
Fix bug in the `is / is not` decision fork.

The program would be incurring in an illegal memory access when trying
to parse formulae such as e.g. `x is`. Greetings to David White for
seeing the crash coming. ^^U
